pennylane/grouping/group_observables.py

```diff
@@ -0,0 +1,252 @@
+
+    Raises:
+        ValueError: if `grouping_type` or `graph_colourer` are not recognized as elements of
+        `GROUPING_TYPES` or `GRAPH_COLOURING_METHODS` respectively.
+
+    """
+
+    def __init__(self, observables, grouping_type="qwc", graph_colourer="rlf"):
+
+        if grouping_type.lower() not in GROUPING_TYPES:
+            raise ValueError(
+                "Grouping type must be one of: {}, instead got {}.".format(
+                    GROUPING_TYPES, grouping_type
+                )
+            )
+
+        self.grouping_type = grouping_type.lower()
+
+        if graph_colourer.lower() not in GRAPH_COLOURING_METHODS.keys():
+            raise ValueError(
+                "Graph colouring method must be one of: {}, instead got {}.".format(
+                    list(GRAPH_COLOURING_METHODS.keys()), graph_colourer
+                )
+            )
+
+        self.graph_colourer = GRAPH_COLOURING_METHODS[graph_colourer.lower()]
+        self.observables = observables
+        self.__wire_map = None
+        self.__n_qubits = None
+        self.binary_observables = None
+        self.adj_matrix = None
+        self.grouped_paulis = None
+
+    def obtain_binary_repr(self, n_qubits=None, wire_map=None):
+        """Converts the list of Pauli words to a binary matrix.
+
+        Keyword args:
+            n_qubits (int): number of qubits to specify dimension of binary vector representation.
+            wire_map (dict): dictionary containing all wire labels used in the Pauli word as keys,
+                and unique integer labels as their values
+
+        Returns:
+            array[bool]: a column matrix of the Pauli words in binary vector representation.
+
+        """
+
+        if wire_map is None:
+            self.__wire_map = {
+                wire: c
+                for c, wire in enumerate(
+                    Wires.all_wires([obs.wires for obs in self.observables]).tolist()
+                )
+            }
+
+        else:
+            self.__wire_map = wire_map
+
+        self.binary_observables = convert_observables_to_binary_matrix(
+            self.observables, n_qubits, self.__wire_map
+        )
+
+        self.__n_qubits = n_qubits
+
+        return self.binary_observables
+
+    def construct_complement_adj_matrix_for_operator(self):
+        """Constructs the adjacency matrix for compliment of Pauli graph.
+
+        The adjacency matrix for an undirected graph of N vertices is a N by N symmetric binary
+        matrix, where matrix elements of 1 denote an edge, and matrix elements of 0 denote no edge.
+
+        Returns:
+            array[bool]: the square and symmetric adjacency matrix.
+
+        """
+
+        if self.binary_observables is None:
+            self.obtain_binary_repr()
+
+        n_qubits = int(np.shape(self.binary_observables)[0] / 2)
+
+        if self.grouping_type == "qwc":
+            adj = get_qwc_compliment_adj_matrix(self.binary_observables)
+
+        else:
```

Comment:
This code path doesn't seem to be used yet; is there a guarantee that all alternative partitioning schemes to QWC will compute the adjacency matric from the symplectic form (say, if more schemes are introduced beyond commuting and anticommuting)?

For the purposes of this pull request, it might be cleaner to stash all the symplectic geometry and introduce it when other partitioning schemes appear.

Reply:
Good point, the use of `else` here was a temporary catch-all for the commuting and anticommuting approaches, will change to `elif`s. 

Right, the use of the binary Pauli vector representation and some of the symplectic tools may appear superfluous with only the QWC approach implemented. I was thinking along the "do more now and less later" strategy, since these symplectic tools are needed for the fully-commuting approach, and it just so happens that some other things are quite convenient to do in the symplectic rep (such as getting the adjacency matrix, for instance), even though other non-symplectic ways exist.